pkg/storage/tpr/install_types.go

```diff
@@ -34,31 +36,66 @@ var thirdPartyResources = []v1beta1.ThirdPartyResource{
 	serviceBindingTPR,
 }
 
+//Installer takes in a client and exposes method for installing third party resources
+type Installer struct {
+	tprs tprs_v1beta1.ThirdPartyResourceInterface
+}
+
+//NewInstaller is used to install third party resources
+func NewInstaller(tprs tprs_v1beta1.ThirdPartyResourceInterface) *Installer {
+	return &Installer{
+		tprs: tprs,
+	}
+}
+
 // InstallTypes installs all third party resource types to the cluster
-func InstallTypes(cs clientset.Interface) error {
-	tprs := cs.Extensions().ThirdPartyResources()
+func (i *Installer) InstallTypes() error {
+	errChans := make(map[string]chan error)
+
 	for _, tpr := range thirdPartyResources {
 		glog.Infof("Checking for existence of %s", tpr.Name)
-		if _, err := tprs.Get(tpr.Name); err == nil {
+		if _, err := i.tprs.Get(tpr.Name); err == nil {
 			glog.Infof("Found existing TPR %s", tpr.Name)
 			continue
 		}
 
 		glog.Infof("Creating Third Party Resource Type: %s", tpr.Name)
-		if _, err := tprs.Create(&tpr); err != nil {
-			glog.Errorf("Failed to create Third Party Resource Type: %s (%s))", tpr.Name, err)
-			return err
-		}
-		glog.Infof("Created TPR '%s'", tpr.Name)
-		// There can be a delay, so poll until it's ready to go...
-		for i := 0; i < 30; i++ {
-			if _, err := tprs.Get(tpr.Name); err != nil {
-				glog.Infof("TPR %s is ready", tpr.Name)
-				break
+		errChans[tpr.Name] = make(chan error, 1)
+		defer close(errChans[tpr.Name])
+
+		go func(tpr v1beta1.ThirdPartyResource, ec chan error) {
+			_, err := i.tprs.Create(&tpr)
+			ec <- err
+		}(tpr, errChans[tpr.Name])
+	}
+
+	errStr := ""
+	for name, ec := range errChans {
```

Comment:
+1 - I commented earlier on how exactly to do this and aggregate the errors. Example code is here: https://play.golang.org/p/2nn2Aqukib. It'll be fairly straightforward to change it to not pipeline all errors through a single channel, which I know was a concern